File: FurnitureStore/FurnitureStore.Services/Database/GiftCard.cs

using FurnitureStore.Services.Domain.Base;

namespace FurnitureStore.Services.Database
{
    public class GiftCard : BaseSoftDeleteEntity
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string CardNumber { get; set; }
        public int Amount { get; set; }
        public DateTime ExpiryDate { get; set; }
        public bool IsActivated { get; set; }
        public virtual ICollection<GiftCardUsers> GiftCardUsers { get; set; } = new List<GiftCardUsers>();
        public string ImagePath { get; set; }
    }
}

File: FurnitureStore/UI/furniturestore_admin/lib/models/gift_card/gift_card.dart
import 'package:json_annotation/json_annotation.dart';

part 'gift_card.g.dart';

@JsonSerializable()
class GiftCardModel {
  int? id;
  String? name;
  String? cardNumber;
  int? amount;
  DateTime? expiryDate;
  bool? isActivated;
  String? imagePath;

  GiftCardModel({
    this.id,
    this.name,
    this.cardNumber,
    this.amount,
    this.expiryDate,
    this.isActivated,
    this.imagePath,
  });

  void updateActivationStatus() {
    if (expiryDate != null && expiryDate!.isBefore(DateTime.now())) {
      isActivated = false;
    }
  }

  factory GiftCardModel.fromJson(Map<String, dynamic> json) =>
      _$GiftCardModelFromJson(json);

  Map<String, dynamic> toJson() => _$GiftCardModelToJson(this);
}
